### PR TITLE
Update catalog-info.yaml for docker publishing

### PR DESCRIPTION
**Pull request by navarone-feekery**
### Related to https://github.com/elastic/crawler/pull/103

Update the `catalog-info.yaml` in a separate PR to try to get the buildkite endpoint to exist.

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -32,3 +35,35 @@ spec:
...
+  owner: "group:ingestion-team"
+  system: "buildkite"
+  implementation:
+    apiVersion: "buildkite.elastic.dev/v1"
+    kind: "Pipeline"
+    metadata:
+      name: "crawler-docker-build-publish"
+    spec:
+      repository: "elastic/crawler"
+      pipeline_file: ".buildkite/release-pipeline.yml"
```
Comment (seanstory) [inline catalog-info.yaml:62]: This file might need to actually exist? But we'll find out. 🤞

**Review by seanstory — DISMISSED** [review]

File: catalog-info.yaml (hunk 1 above)
Comment (acrewdson) [reply catalog-info.yaml:62]: pretty sure this won't work without that file, but could be wrong

maybe easiest to merge https://github.com/elastic/crawler/pull/103 even if it's not complete/done and then iterate on it from there

Comment (navarone-feekery) [reply catalog-info.yaml:62]: I think I'll just focus on #103, as I don't want to spend more effort on this PR if it's more than just catalog-info.yml.
I'm sure something will be automatically spun up for us (and if not, we can troubleshoot from there)

**Comment by navarone-feekery** [comment]
Closing in favour of including the catalog-info.yaml in https://github.com/elastic/crawler/pull/103

File: catalog-info.yaml (hunk 1 above)
Comment (seanstory) [reply catalog-info.yaml:62]: I think we could have merged this as it was, and/or could have added a `.buildkite/release-pipeline.yml` with a single step of `echo 'hello world'`.

The bummer about not having this merged is that you can't test the pipeline in 103 without merging it first. Which is generally not how we like to do things. I don't think there's much risk in this, so Nav if you're fine merging 103 _whenever_, then trying it out and fixing issues in a follow-up PR, that's fine. If you'd rather all the real changes be in one PR, I think you might want to come back to this approach. Either is fine.

Comment (navarone-feekery) [reply catalog-info.yaml:62]: Yeah that makes sense. No idea what I was thinking yesterday. I think I was just trying to reduce PRs.

**Review by jedrazb — APPROVED** [review]

**Comment by navarone-feekery** [comment]
buildkite test this

**Comment by navarone-feekery** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|0.1|[](https://github.com/elastic/crawler/pull/107)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**